codepropertygraph/src/main/scala/io/shiftleft/utils/StringInterner.scala

```diff
@@ -0,0 +1,34 @@
+package io.shiftleft.utils
+
+import gnu.trove.map.hash.THashMap
+
+trait StringInterner {
+  def intern(s: String): String
+}
+
+object StringInterner {
+  val DefaultMaxStringLength: Int = 1024
+  val DefaultInitialSize: Int = 64 * 1024
+
+  val noop: StringInterner = new StringInterner {
+    override def intern(s: String): String = s
+  }
+  def makeStrongInterner(maxStringLength: Int = DefaultMaxStringLength,
+                         initialSize: Int = DefaultInitialSize): StringInterner = new StringInterner {
+    private val stringCache = new THashMap[String, String](initialSize)
+
+    def intern(s: String): String = {
+      if (s.length < maxStringLength) {
```

Comment:
Why do we not intern long strings? Wouldnt we save most memory for those?

Reply:
@ml86 Looking at heap dump I've found that almost all of the savings come from small-medium strings, like java method signatures so I've decided to limit somehow the amount of memory that it holds on to.